PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/ManifestFileUtilsEx.java

```diff
@@ -0,0 +1,100 @@
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.hdinsight.spark.common;
+
+import com.intellij.ide.util.ClassFilter;
+import com.intellij.ide.util.TreeClassChooser;
+import com.intellij.ide.util.TreeClassChooserFactory;
+import com.intellij.ide.util.TreeJavaClassChooserDialog;
+import com.intellij.openapi.application.ReadAction;
+import com.intellij.openapi.application.WriteAction;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.search.GlobalSearchScope;
+import com.microsoft.azure.hdinsight.spark.ui.SparkSubmissionContentPanelConfigurable;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.azurecommons.helpers.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+
+public class ManifestFileUtilsEx {
+    @Nullable
+    public static PsiClass selectMainClass(@NotNull Project project,
+                                    @Nullable VirtualFile jarFile,
+                                    final @NotNull SparkSubmissionContentPanelConfigurable configurable) {
+        if (jarFile == null) {
+            return null;
+        }
+
+        final TreeClassChooserFactory chooserFactory = TreeClassChooserFactory.getInstance(project);
+        final GlobalSearchScope searchScope = GlobalSearchScope.everythingScope(project);
+        // TODO: the following code is used to find initialClassName in the jar. When user specified initialClassName
+        // in the main-class-selection textfield and clicked the main-class-selection button, the filter result in the dialog
+        // should be the initialClass. Currently we don't enable this method since exception happens with the following code.
+        // final PsiClass aClass = initialClassName != null ? JavaPsiFacade.getInstance(project).findClass(initialClassName, searchScope) : null;
+        final TreeClassChooser chooser =
+                chooserFactory.createWithInnerClassesScopeChooser("Select Main Class", searchScope, new JarClassFilter(jarFile.getPath()), null);
+
+        ((TreeJavaClassChooserDialog) chooser).getWindow().addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowOpened(WindowEvent e) {
+                // remove old jar and add new jar to project dependency
+                WriteAction.runAndWait(() -> configurable.removeOldJarAndAddNewJarToProjectLibrary());
```

Comment:
I suppose needn't to remove the old jars, is it?

Reply:
Currently we will remove the jar from module dependency **when we close the run/configuration dialog**, so it's necessary to remove old jar each time user click the main class selection button. Why don't we remove the jar and module **when we close the main class selection dialog**? Since in that case, if user click the main class selection button several times without changing the jar path, we will remove and add dependency frequently, which is not a efficient choice.